### PR TITLE
Attributes all

### LINKED ISSUE #689: Mesh.face_attributes does not return all attributes
**Describe the bug**
`Mesh.face_attributes(fkey)` only returns the default attributes that have been set using `Mesh.update_default_face_attributes()`.

**To Reproduce**
Run the code bellow:
```
from compas.geometry import Box
from compas.datastructures import Mesh

box = Box.from_width_height_depth(2, 2, 2)
box = Mesh.from_shape(box)
box.update_default_face_attributes({"attr1":"value1", "attr2":"value2"})

random_fkey = box.get_any_face()
print(random_fkey, box.face_attributes(random_fkey))

box.face_attribute(random_fkey, "attr3", "value3")
print(random_fkey, box.face_attributes(random_fkey))
print(random_fkey, box.face_attribute(random_fkey, "attr3"))
```

**Expected behavior**
Return all attributes, including the ones that have been assigned only to that face

**Desktop (please complete the following information):**
 - Compas: 0.18.1
 - Python version: 3.7.6
 - Python package manager: conda

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomvanmele** [comment]
the problem lies in the `__iter__` method of `AttributeView`. it only looks at the defaults even though all of the other methods look in both. as a result all mapping methods of `MutableMapping` return key, items, values only from the defaults dict...

**Comment by tomvanmele** [comment]
not sure if the proposed solution is the simplest possible though...

File: src/compas/datastructures/attributes.py [hunk 1]
```diff
@@ -48,43 +48,44 @@ def __iter__(self):
             for name in self.attr:
                 yield name
         else:
-            for name in self.defaults:
+            names = set(self.defaults).union(self.attr)
+            for name in names:
```
Comment (beverlylytle) [inline src/compas/datastructures/attributes.py:52]: Should `__len__` reflect this change?

Comment (tomvanmele) [reply src/compas/datastructures/attributes.py:52]: yes indeed it should

**Comment by tomvanmele** [comment]
@gonzalocasas ping

**Comment by gonzalocasas** [comment]
lgtm! if build is green, go for it

**Comment by gonzalocasas** [comment]
@brgcode merge?

**Pull request merged**